apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/FakeScheduler.cs

```diff
@@ -312,6 +311,12 @@ public async Task RunAndPause(TimeSpan time)
                     // Release all due, non-cancelled timers. (A cancelled TCS will ignore this.)
                     timers.ForEach(t => t.CompletionSource.TrySetResult(0));
 
+                    // Give a little time for things to run before we enter the monitor at all.
+                    // On CI platforms, we've seen odd behaviour where tasks appear to be effectively
+                    // blocked by this thread. Using a Task.Delay instead of Thread.Sleep will allow
+                    // other tasks (e.g. those unblocked by TCS.TrySetResult) a better chance to run.
+                    await Task.Delay(IdleTimeBeforeAdvancing).ConfigureAwait(false);
```

Comment:
In theory at least, this thread doesn't actually block other threads. We don't own the monitor while Monitor.Wait is executing, so other threads should be able to get in. I believe the problem in flakiness is only because of the way the TPL is scheduling tasks. By this point, if they've woken up and are running but haven't yet scheduled anything, they still *can* schedule something.

Note that we want to be able to run for a total time of longer than IdleTimeBeforeAdvancing, if we're creating multiple timers for whatever reason, so I think we do want a loop. Ideally we'd use a non-blocking approach to this, but I'm not sure of the best way of doing that. (We could create another TCS to indicate that a task has been created, and set the result of that from the AddDelay method? I can see all kinds of race conditions there though.)